ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/PeerSimulator/Run.hs

```diff
@@ -288,12 +252,22 @@ runPointSchedule k asc pointSchedule tracer blockTree =
     let getCandidates = traverse readTVar candidates
     PeerSimulator.BlockFetch.startBlockFetchLogic registry chainDb fetchClientRegistry getCandidates
     runScheduler tracer pointSchedule resources
-    res <- killAll (Map.elems chainSyncThreads)
+    chainSyncExceptions <- collectExceptions (Map.elems chainSyncRess)
     b <- atomically $ ChainDB.getCurrentChain chainDb
-    pure $ maybe (Right b) Left res
+    pure $ maybe (Right b) Left chainSyncExceptions
   where
     config = defaultCfg k
 
+    collectExceptions :: [StrictTVar m (Maybe SomeException)] -> m (Maybe (NonEmpty SomeException))
```

Comment:
I think it would be great if we could bundle the `PeerId` with the exception. today we got a `TimeoutExceeded` and could have used the info in the message!

Reply:
Done in d6a6e87

Reply:
awesome!